include/envoy/network/listen_socket.h

```diff
@@ -154,8 +163,8 @@ class ConnectionSocket : public virtual Socket {
    *        by iptables REDIRECT. The caller is responsible for making sure the new address is
    *        actually different when passing restored as 'true'.
    */
-  virtual void setLocalAddress(const Address::InstanceConstSharedPtr& local_address,
-                               bool restored) PURE;
+  virtual void setConnectionLocalAddress(const Address::InstanceConstSharedPtr& local_address,
+                                         bool restored) PURE;
```

Comment:
IMO this renaming is not necessary. I'd prefer keeping the function named as `setLocalAddress()`.

Or maybe even better, due to the new function in the base class covering the case where the local address is set without touching the `local_address_resored_` member, maybe change this to `restoreLocalAddress(local_address)` without the explicit boolean parameter and change the current callers using `false` use the base class `setLocalAddress()` and the current callers using `true` use this (`restoreLocalAddress()`), which always sets `local_address_restored_` as `true`. This change loses the capability to flip `local_address_restored_` back to `false' once restored, but we don't need that capability.

Reply:
I definitely prefer the restoreLocalAddress. I was originally going to do that, but I was concerned about the flip back to false functionality. If we don't need that, then I'm good for this change.